README.md

```diff
@@ -2,6 +2,118 @@
 
 Option is a C# representation of the maybe monad
 
+### Monad
+Monads are used to encapsulate a value and simplify handling of unusual cases. You can think of monads as generic types.
+
+### Maybe
+Maybe is a data type (also known as an option type) which is used to represent an optional value.
+
+### Maybe Monad
+Maybe monad is a generic type which contains either some value of it's underlying type or nothing.
+
+If you need more information you can check [wikibooks introduction](https://en.wikibooks.org/wiki/Haskell/Understanding_monads/Maybe) to maybe monad.
+
+# Why we need the Option library?
+
+> For simplicity you can think of `Nullable<T>` being closest thing to the maybe monad in C#. Although it doesn't represent all the characteristics of the maybe monad, it's sufficient enough to give the main idea.
+
+At first glance, extending `Nullable<T>` might be a good option, 
+but the `struct` constraint for `T` in it makes it impossible to use with the reference types.
+
+Using maybe monad also makes code more readable by reducing the nested code blocks, as you will see in the following examples.
+
+# How to use it
```

Comment:
I would consider moving this paragraph up to the top of the README.
I believe the first thing people want to see is the code because that answers so many questions.

* What imports do I need?
* Is the syntax nice?
* Do I need a maths degree to use this library?
* Will it make my code more or less maintainable?